Grafana Upgrade Failed

Grafana Helm Chart 7.1.0
2024-01-10T17:33:47.568445292Z Error: UPGRADE FAILED: template: grafana/templates/secret.yaml:1:62: executing "grafana/templates/secret.yaml" at <.Values.env.GF_SECURITY_ADMIN_PASSWORD__FILE>: can't evaluate field GF_SECURITY_ADMIN_PASSWORD__FILE in type interface {}

We use OAuth for authentication and do not use local admin/access accounts

admin:
  existingSecret: ''
  passwordKey: admin-password
  userKey: admin-user
adminUser: admin
affinity: {}
alerting: {}
assertNoLeakedSecrets: false
autoscaling:
  enabled: false
  minReplicas: 1
  maxReplicas: 5
  targetCPU: "60"
  targetMemory: ""
  behavior: {}
containerSecurityContext:
  allowPrivilegeEscalation: false
  capabilities:
    drop:
      - ALL
  seccompProfile:
    type: RuntimeDefault
createConfigmap: true
dashboardProviders: {}
dashboards: {}
dashboardsConfigMaps: {}
datasources: {}
deploymentStrategy:
  type: RollingUpdate
dnsConfig: {}
dnsPolicy: null
downloadDashboards:
  env: {}
  envFromSecret: ''
  envValueFrom: {}
  resources: {}
  securityContext:
    allowPrivilegeEscalation: false
    capabilities:
      drop:
        - ALL
    seccompProfile:
      type: RuntimeDefault
downloadDashboardsImage:
  pullPolicy: IfNotPresent
  registry: docker.io
  repository: curlimages/curl
  sha: ''
  tag: 7.85.0
enableKubeBackwardCompatibility: false
enableServiceLinks: true
env:
  - name: HTTP_HOST
    value: "0.0.0.0"
  - name: POD_IP
    valueFrom:
      fieldRef:
        fieldPath: status.podIP
  - name: GRAFANA_DB_PASSWORD
    valueFrom:
      configMapKeyRef:
        name: grafana-config
        key: GRAFANA_DB_PASSWORD
  - name: CLIENT_ID
    valueFrom:
      configMapKeyRef:
        name: grafana-config
        key: CLIENT_ID
  - name: CLIENT_SECRET
    valueFrom:
      configMapKeyRef:
        name: grafana-config
        key: CLIENT_SECRET
envFromConfigMaps: {}
envFromSecret: ''
envFromSecrets: []
envRenderSecret: {}
envValueFrom: {}
extraConfigmapMounts: []
extraContainerVolumes: []
extraContainers: ''
extraEmptyDirMounts: []
extraExposePorts: []
extraInitContainers: []
extraLabels: {}
extraObjects: []
extraSecretMounts: []
extraVolumeMounts: []
extraVolumes: []
global:
  imagePullSecrets: []
  imageRegistry: null
  cattle:
    systemProjectId: p-dkrnd
gossipPortName: gossip
grafana.ini:
  analytics:
    check_for_updates: true
  grafana_net:
    url: https://grafana.net
  log:
    mode: console
  paths:
    data: /var/lib/grafana/
    logs: /var/log/grafana
    plugins: /var/lib/grafana/plugins
    provisioning: /etc/grafana/provisioning
  server:
    domain: grafana.thetmg.com
    enforce_domain: true
    root_url: https://grafana.thetmg.com
  auth.anonymous:
    enabled: true
    org_name: Tarbell Management Group
    org_role: Admin
  auth.azuread:
    allow_assign_grafana_admin: true
    allow_sign_up: true
    auth_url: >-
      https://login.microsoftonline.com/1fbd3e81-9940-428f-8ba6-463db4dcdf62/oauth2/v2.0/authorize
    auto_login: true
    client_id: ${CLIENT_ID}
    client_secret: ${CLIENT_SECRET}
    enabled: true
    role_attribute_strict: false
    scopes: openid email profile
    skip_org_role_sync: false
    token_url: >-
      https://login.microsoftonline.com/1fbd3e81-9940-428f-8ba6-463db4dcdf62/oauth2/v2.0/token
    use_pkce: true
  database:
    host: mysql-1699562096.mysql.svc.cluster.local:3306
    name: grafana
    password: ${GRAFANA_DB_PASSWORD}
    type: mysql
    user: grafana
  feature_toggles:
    correlations: true
    dashgpt: true
    extraThemes: true
    nestedFolders: true
headlessService: false
hostAliases: []
image:
  pullPolicy: IfNotPresent
  pullSecrets: []
  registry: docker.io
  repository: grafana/grafana-enterprise
  sha: ''
  tag: latest
imageRenderer:
  affinity: {}
  autoscaling:
    behavior: {}
    enabled: false
    maxReplicas: 5
    minReplicas: 1
    targetCPU: '60'
    targetMemory: ''
  containerSecurityContext:
    allowPrivilegeEscalation: false
    capabilities:
      drop:
        - ALL
    readOnlyRootFilesystem: true
    seccompProfile:
      type: RuntimeDefault
  deploymentStrategy: {}
  enabled: false
  env: {}
  envValueFrom: {}
  grafanaProtocol: http
  grafanaSubPath: ''
  hostAliases: []
  image:
    pullPolicy: Always
    registry: docker.io
    repository: grafana/grafana-image-renderer
    sha: ''
    tag: latest
  networkPolicy:
    extraIngressSelectors: []
    limitEgress: false
    limitIngress: true
  nodeSelector: {}
  podAnnotations: {}
  podPortName: http
  priorityClassName: ''
  replicas: {}
  resources: {}
  revisionHistoryLimit: 10
  securityContext: {}
  service:
    appProtocol: ''
    enabled: true
    port: 8081
    portName: http
    targetPort: 8081
  serviceAccountName: ''
  serviceMonitor:
    enabled: false
    interval: 1m
    labels: {}
    path: /metrics
    relabelings: []
    scheme: http
    scrapeTimeout: 30s
    targetLabels: []
    tlsConfig: {}
  tolerations: []
ingress:
  annotations: {}
  enabled: true
  extraPaths: []
  hosts:
    - grafana.thetmg.com
  labels: {}
  path: /
  pathType: Prefix
  tls:
    - hosts:
        - grafana.thetmg.com
      secretName: thetmg
  ingressClassName: nginx
initChownData:
  enabled: true
  image:
    pullPolicy: IfNotPresent
    registry: docker.io
    repository: library/busybox
    sha: ''
    tag: 1.31.1
  resources: {}
  securityContext:
    capabilities:
      add:
        - CHOWN
    runAsNonRoot: false
    runAsUser: 0
    seccompProfile:
      type: RuntimeDefault
ldap:
  config: ''
  enabled: false
  existingSecret: ''
lifecycleHooks: {}
livenessProbe:
  failureThreshold: 10
  httpGet:
    path: /api/health
    port: 3000
  initialDelaySeconds: 60
  timeoutSeconds: 30
namespaceOverride: ''
networkPolicy:
  allowExternal: true
  egress:
    blockDNSResolution: false
    enabled: false
    ports: []
    to: []
  enabled: false
  explicitNamespacesSelector: {}
  ingress: true
nodeSelector: {}
notifiers: {}
persistence:
  accessModes:
    - ReadWriteOnce
  enabled: false
  extraPvcLabels: {}
  finalizers:
    - kubernetes.io/pvc-protection
  inMemory:
    enabled: false
  size: 10Gi
  type: pvc
plugins:
  - agenty-flowcharting-panel
  - grafana-clock-panel
  - flant-statusmap-panel
  - influxdata-flightsql-datasource
  - marcusolsson-json-datasource
podDisruptionBudget:
  enabled: false
  minAvailable: 2
podPortName: grafana
rbac:
  create: true
  extraClusterRoleRules: []
  extraRoleRules: []
  namespaced: false
  pspEnabled: false
  pspUseAppArmor: false
readinessProbe:
  httpGet:
    path: /api/health
    port: 3000
replicas: {}
resources:
  limits:
    cpu: 250m
    memory: 256Mi
  requests:
    cpu: 100m
    memory: 128Mi
revisionHistoryLimit: 10
securityContext:
  fsGroup: 472
  runAsGroup: 472
  runAsNonRoot: true
  runAsUser: 472
service:
  annotations: {}
  appProtocol: ''
  enabled: true
  labels: {}
  port: 80
  portName: service
  targetPort: 3000
  type: ClusterIP
serviceAccount:
  autoMount: true
  create: true
  labels: {}
  name: null
  nameTest: null
serviceMonitor:
  enabled: false
  interval: 30s
  labels: {}
  metricRelabelings: []
  path: /metrics
  relabelings: []
  scheme: http
  scrapeTimeout: 30s
  targetLabels: []
  tlsConfig: {}
sidecar:
  alerts:
    enabled: false
    env: {}
    extraMounts: []
    initAlerts: false
    label: grafana_alert
    labelValue: ''
    reloadURL: https://grafana.thetmg.com/api/admin/provisioning/alerting/reload
    resource: both
    script: null
    searchNamespace: null
    sizeLimit: {}
    skipReload: false
    watchMethod: WATCH
  dashboards:
    SCProvider: true
    defaultFolderName: null
    enabled: false
    env: {}
    extraMounts: []
    folder: /tmp/dashboards
    folderAnnotation: null
    label: grafana_dashboard
    labelValue: ''
    provider:
      allowUiUpdates: false
      disableDelete: false
      folder: ''
      foldersFromFilesStructure: false
      name: sidecarProvider
      orgid: 1
      type: file
    reloadURL: http://localhost:3000/api/admin/provisioning/dashboards/reload
    resource: both
    script: null
    searchNamespace: null
    sizeLimit: {}
    skipReload: false
    watchMethod: WATCH
  datasources:
    enabled: false
    env: {}
    initDatasources: false
    label: grafana_datasource
    labelValue: ''
    reloadURL: http://localhost:3000/api/admin/provisioning/datasources/reload
    resource: both
    script: null
    searchNamespace: null
    sizeLimit: {}
    skipReload: false
    watchMethod: WATCH
  enableUniqueFilenames: false
  image:
    registry: quay.io
    repository: kiwigrid/k8s-sidecar
    sha: ''
    tag: 1.25.2
  imagePullPolicy: IfNotPresent
  livenessProbe: {}
  notifiers:
    enabled: false
    env: {}
    initNotifiers: false
    label: grafana_notifier
    labelValue: ''
    reloadURL: http://localhost:3000/api/admin/provisioning/notifications/reload
    resource: both
    script: null
    searchNamespace: null
    sizeLimit: {}
    skipReload: false
    watchMethod: WATCH
  plugins:
    enabled: false
    env: {}
    initPlugins: false
    label: grafana_plugin
    labelValue: ''
    reloadURL: http://localhost:3000/api/admin/provisioning/plugins/reload
    resource: both
    script: null
    searchNamespace: null
    sizeLimit: {}
    skipReload: false
    watchMethod: WATCH
  readinessProbe: {}
  resources: {}
  securityContext:
    allowPrivilegeEscalation: false
    capabilities:
      drop:
        - ALL
    seccompProfile:
      type: RuntimeDefault
smtp:
  existingSecret: ''
  passwordKey: password
  userKey: user
testFramework:
  enabled: true
  image:
    registry: docker.io
    repository: bats/bats
    tag: v1.4.1
  imagePullPolicy: IfNotPresent
  securityContext: {}
tolerations: []
topologySpreadConstraints: []
useStatefulSet: false

@bgatestmg same for me.
when deploying the chart with my local maschine its working, but on my Git Runner i get the similiar error

`Error: UPGRADE FAILED: execution error at (grafana/templates/deployment.yaml:36:28): Sensitive key 'database.password' should not be defined explicitly in values. Use variable expansion instead.
Error: Process completed with exit code 1.`

i was running my .yaml with the `envFromSecret` stanza and changed for debugging to the `extraSecretMounts` stanza as in the official Documentation is described.
Both ways are working locally but end in an error with Ci/CI

--- a/charts/grafana/Chart.yaml
+++ b/charts/grafana/Chart.yaml
@@ -1,6 +1,6 @@
 apiVersion: v2
 name: grafana
-version: 7.2.1
+version: 7.2.2
 appVersion: 10.2.3
 kubeVersion: "^1.8.0-0"
 description: The leading tool for querying and visualizing time series and metrics.

--- a/charts/grafana/templates/_helpers.tpl
+++ b/charts/grafana/templates/_helpers.tpl
@@ -263,7 +263,9 @@ sensitiveKeys:
         {{- range $index, $elem := $secret.path -}}
           {{- if and $shouldContinue (hasKey $currentMap $elem) -}}
             {{- if eq (len $secret.path) (add1 $index) -}}
-              {{- fail (printf "Sensitive key '%s' should not be defined explicitly in values. Use variable expansion instead." (join "." $secret.path)) -}}
+              {{- if not (regexMatch "\\$(?:__(?:env|file|value))?{[^}]+}" (index $currentMap $elem)) -}}
+                {{- fail (printf "Sensitive key '%s' should not be defined explicitly in values. Use variable expansion instead." (join "." $secret.path)) -}}
+              {{- end -}}
             {{- else -}}
               {{- $currentMap = index $currentMap $elem -}}
             {{- end -}}

--- a/charts/grafana/templates/_helpers.tpl
+++ b/charts/grafana/templates/_helpers.tpl
@@ -263,7 +263,7 @@ sensitiveKeys:
         {{- range $index, $elem := $secret.path -}}
           {{- if and $shouldContinue (hasKey $currentMap $elem) -}}
             {{- if eq (len $secret.path) (add1 $index) -}}
-              {{- if not (regexMatch "\\$(?:__(?:env|file|value))?{[^}]+}" (index $currentMap $elem)) -}}
+              {{- if not (regexMatch "\\$(?:__(?:env|file|vault))?{[^}]+}" (index $currentMap $elem)) -}}
                 {{- fail (printf "Sensitive key '%s' should not be defined explicitly in values. Use variable expansion instead." (join "." $secret.path)) -}}
               {{- end -}}
             {{- else -}}

--- a/charts/grafana/templates/_helpers.tpl
+++ b/charts/grafana/templates/_helpers.tpl
@@ -264,7 +264,7 @@ sensitiveKeys:
           {{- if and $shouldContinue (hasKey $currentMap $elem) -}}
             {{- if eq (len $secret.path) (add1 $index) -}}
               {{- if not (regexMatch "\\$(?:__(?:env|file|vault))?{[^}]+}" (index $currentMap $elem)) -}}
-                {{- fail (printf "Sensitive key '%s' should not be defined explicitly in values. Use variable expansion instead." (join "." $secret.path)) -}}
+                {{- fail (printf "Sensitive key '%s' should not be defined explicitly in values. Use variable expansion instead. You can disable this client-side validation by changing the value of assertNoLeakedSecrets." (join "." $secret.path)) -}}
               {{- end -}}
             {{- else -}}
               {{- $currentMap = index $currentMap $elem -}}

